### DISCUSSION TITLE
Scoped API Key

### DISCUSSION THREAD

**Discussion by duronald** [mrrfv/cloudflare-gateway-pihole-scripts#21] [Q&A] [answered]
Hi,

What scoped API key permissions did you use in Cloudflare for your key? I tried setting the permissions to allow read edit and report of Zero Trust but it isn't working.

Thank you

**Comment by mrrfv** [comment]
You want a Cloudflare Global API key, not API token. They are vastly different from each other (the API key has no permissions to set).

![image](https://github.com/mrrfv/cloudflare-gateway-pihole-scripts/assets/92185687/0151d8dd-eebb-4289-a6ac-e939bddc25fe)

**Comment by duronald** [comment, marked as answer]
I didn’t use a global API key because it’s a pretty big security risk. It doesn’t seem like Cloudflare Gateway supports scoped API tokens yet so I created a new child account that only has Zero Trust and Gateway access and used the global key from the child account.
